mls-rs/src/group/mod.rs

```diff
@@ -602,6 +604,155 @@ where
+        plaintext: &[u8],
+    ) -> Result<HpkeCiphertext, MlsError> {
+        let component_operation_label =
+            ComponentOperationLabel::new(component_id, context.to_vec());
+        self.hpke_encrypt_to_recipient_with_generic_context(
+            recipient_index,
+            &component_operation_label.get_bytes()?,
+            associated_data,
+            plaintext,
+        )
+        .await
+    }
+
+    #[cfg_attr(not(mls_build_async), maybe_async::must_be_sync)]
+    async fn hpke_decrypt_for_current_member_with_generic_context(
+        &self,
+        context_info: &[u8],
+        associated_data: Option<&[u8]>,
+        hpke_ciphertext: HpkeCiphertext,
+    ) -> Result<Vec<u8>, MlsError> {
+        let self_private_key = &self.private_tree.secret_keys[0]
+            .as_ref()
+            .ok_or(MlsError::InvalidTreeKemPrivateKey)?;
+        let self_public_key = &self.current_user_leaf_node()?.public_key;
+        let plaintext = self
+            .cipher_suite_provider
+            .hpke_open(
+                &hpke_ciphertext,
+                self_private_key,
+                self_public_key,
+                context_info,
+                associated_data,
+            )
+            .await
+            .map_err(|e| MlsError::CryptoProviderError(e.into_any_error()))?;
+        Ok(plaintext)
+    }
+
+    /// HPKE decrypts a message sent to the current member.
+    ///
+    /// Takes `HpkeCiphertext` generated by `hpke_encrypt_to_recipient` intended for the
+    /// current member.
+    ///
+    /// WARNING: The message sender is not authenticated.
+    #[cfg(all(feature = "non_domain_separated_hpke_encrypt_decrypt", feature = "ffi"))]
+    #[cfg_attr(
+        not(mls_build_async),
+        // all(feature = "ffi", not(test)),
+        maybe_async::must_be_sync,
+        safer_ffi_gen::safer_ffi_gen_ignore
+    )]
+    pub async fn hpke_decrypt_for_current_member(
+        &self,
+        context_info: &[u8],
+        associated_data: Option<&[u8]>,
+        hpke_ciphertext: HpkeCiphertext,
+    ) -> Result<Vec<u8>, MlsError> {
+        self.hpke_decrypt_for_current_member_with_generic_context(
+            context_info,
+            associated_data,
+            hpke_ciphertext,
+        )
+        .await
+    }
+
+    /// HPKE decrypts a message sent to the current member.
+    ///
+    /// Takes `HpkeCiphertext` generated by `hpke_encrypt_to_recipient` intended for the
+    /// current member.
+    #[cfg_attr(not(mls_build_async), maybe_async::must_be_sync)]
+    pub async fn safe_decrypt_with_context_for_current_member(
+        &self,
+        component_id: ComponentID,
+        context: &[u8],
+        associated_data: Option<&[u8]>,
+        hpke_ciphertext: HpkeCiphertext,
+    ) -> Result<Vec<u8>, MlsError> {
+        let component_operation_label =
+            ComponentOperationLabel::new(component_id, context.to_vec());
+        self.hpke_decrypt_for_current_member_with_generic_context(
```

Comment:
To make it a bit more efficient, I think that you can avoid the `to_vec` call if you define `ComponentOperationLabel` as follows (I don't think `MlsDecode` is needed for anything)
```
#[derive(Clone, Debug, PartialEq, MlsSize, MlsEncode)]
pub struct ComponentOperationLabel<'a> {
    label: &'static [u8],
    component_id: ComponentID,
    context: 'a[u8],
}
```

Reply:
Good idea; done!